packages/@aws-cdk/aws-apigatewayv2/lib/http/authorizer.ts

```diff
@@ -139,7 +216,7 @@ export interface HttpRouteAuthorizerBindOptions {
   /**
    * The scope for any constructs created as part of the bind.
    */
-  readonly scope: Construct;
+  readonly scope: CoreConstruct;
```

Comment:
why is this being changed?

Reply:
Because lambda.addPermission requires scope to be of this type. See https://github.com/aws/aws-cdk/pull/13181/files/ea51b9d693de5a6bfcf97bb882706b30fd78e975#diff-2c8f1a771d6f62decf7e5ef20ebf6de6218c25ade0f1531ed5e78609f3bc23bdR105


TLDR: the construct struct needed the by `scope` parameter of lambda.addPermission is CoreConstruct and not Construct

Reply:
With CDKv2, `CoreConstruct` will be gone.

Instead, can you do something like -

```ts
      this.props.handler.addPermission(`${Names.nodeUniqueId(this.authorizer.node)}-Permission`, {
        scope: options.scope as CoreConstruct,
```